js/clinicalTimeline.js

```diff
@@ -782,27 +786,37 @@ window.clinicalTimeline = (function(){
       var dayFormat = [];
       var m;
       var d;
-      if (time.y !== 0) {
-        dayFormat = dayFormat.concat(time.y+"y");
-      }
-      if (time.m !== 0) {
-        if (time.y !== 0) {
-          m = Math.abs(time.m);
-        } else {
-          m = time.m;
-        }
-        dayFormat = dayFormat.concat(m+"m");
-      }
-      if (time.d !== 0) {
-        if (time.y !== 0 || time.m !== 0) {
-          d = Math.abs(time.d);
-        } else {
-          d = time.d;
-        }
-        dayFormat = dayFormat.concat(d+"d");
-      }
+      var daysPerYear = 365;
+      var daysPerMonth = 30;
       if (time.y === 0 && time.m === 0 && time.d === 0) {
         dayFormat = [0];
+      } else {
+        if (getZoomLevel(beginning,ending,width) === "days" || getZoomLevel(beginning,ending,width) === "10days" || getZoomLevel(beginning,ending,width) === "3days") {
+          if (time.y !== 0 && time.m !== 0) {
+            d = time.m * daysPerMonth + time.y * daysPerYear + time.d;
+          } else if (time.y == 0 && time.m !== 0) {
+            d = time.m * daysPerMonth + time.d;
+          } else {
+            d = time.d;
+          }
+          dayFormat = dayFormat.concat(m + "d");
```

Comment:
This part looks a little too complicated. I think you can use time.toDays() and concat the "d"